tests/test_misc.py

```diff
@@ -1908,6 +1909,113 @@ def test_spends_conflicting(revault_network, bitcoind):
     )
 
 
+@pytest.mark.skipif(not POSTGRES_IS_SETUP, reason="Needs Postgres for servers db")
+def test_spend_threshold(revault_network, bitcoind, executor):
+    CSV = 2016
+    managers_threshold = 3
+    revault_network.deploy(17, 8, csv=CSV, managers_threshold=managers_threshold)
+    man = revault_network.man(0)
+
+    # Get some more funds
+    bitcoind.generate_block(1)
+
+    vaults = []
+    deposits = []
+    deriv_indexes = []
+    total_amount = 0
+    for i in range(10):
+        amount = random.randint(5, 5000) / 100
+        vaults.append(revault_network.fund(amount))
+        deposits.append(f"{vaults[i]['txid']}:{vaults[i]['vout']}")
+        deriv_indexes.append(vaults[i]["derivation_index"])
+        total_amount += vaults[i]["amount"]
+    revault_network.activate_fresh_vaults(vaults)
+
+    feerate = 1
+    n_outputs = random.randint(1, 3)
```

Comment:
We can also get rid of the randomness, i'm not convinced it was a good idea when i did it for test_large_spend, but for this test i see no reason really